### PR TITLE
Feature/crmsh implementation

### PR DESCRIPTION
**Pull request by arbulu89**
Create salt crmshmod salt module and state code to provide crmsh core functionalities

### CODE REVIEW

**Comment by ayoub-belarbi** [comment]
Well done @arbulu89, looks of for me :)

File: salt/states/crmshmod.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        execute the command in quiet mode (no output)
+    """
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    if __salt__['crm.status']():
+        ret['result'] = True
+        ret['comment'] = 'Cluster is not running already'
```
Comment (diegoakechi) [inline salt/states/crmshmod.py]: Maybe I didn't understand the logic here. If the result of `crm status` is true, doesn't it mean that the cluster is still running?
I would also change the message to:

```suggestion
        ret['comment'] = 'Cluster is already not running'
```

File: salt/states/crmshmod.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        return ret
+
+    if __opts__['test']:
+        ret['result'] = None
+        ret['comment'] = 'Cluster in {} would be removed'.format(name)
```
Comment (diegoakechi) [inline salt/states/crmshmod.py]:
```suggestion
        ret['comment'] = 'Cluster node {} would be removed'.format(name)
```

File: salt/states/crmshmod.py [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    try:
+        #  Here starts the actual process
+        result = __salt__['crm.cluster_remove'](
+            host=name,
+            force=True,
+            quiet=quiet)
+
+        if result:
+            ret['changes']['name'] = name
+            ret['comment'] = 'Error removing cluster'
```
Comment (diegoakechi) [inline salt/states/crmshmod.py]:
```suggestion
            ret['comment'] = 'Error removing cluster node'
```

File: salt/states/crmshmod.py [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+            ret['result'] = False
+            return ret
+
+        ret['changes']['name'] = name
+        ret['comment'] = 'Cluster removed'
```
Comment (diegoakechi) [inline salt/states/crmshmod.py]:
```suggestion
        ret['comment'] = 'Cluster node removed'
```

File: salt/states/crmshmod.py [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+            the sbd is set as diskless.
+    quiet:
+        execute the command in quiet mode (no output)
+    """
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    if not __salt__['crm.status']():
```
Comment (diegoakechi) [inline salt/states/crmshmod.py]: As mentioned above, the logic seems inverted for me for the status call...

File: salt/states/crmshmod.py [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+        #  Here starts the actual process
+        result = __salt__['crm.cluster_join'](
+            host=name,
+            watchdog=watchdog,
+            interface=interface,
+            quiet=quiet)
+
+        if result:
+            ret['changes']['name'] = name
+            ret['comment'] = 'Error joining to cluster'
```
Comment (diegoakechi) [inline salt/states/crmshmod.py]:
```suggestion
            ret['comment'] = 'Error joining to the cluster'
```

File: salt/states/crmshmod.py [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+            ret['result'] = False
+            return ret
+
+        ret['changes']['name'] = name
+        ret['comment'] = 'Node joined to cluster'
```
Comment (diegoakechi) [inline salt/states/crmshmod.py]:
```suggestion
        ret['comment'] = 'Node joined to the cluster'
```

File: salt/states/crmshmod.py (hunk 1 above)
Comment (arbulu89) [reply salt/states/crmshmod.py]: Yeah, maybe it's not totally clear how it works...
Basically, __salt__['crm.status'} returns the returncode of the command crm status. So, if the code is 0, everything went fine, ofherwise some error happened.

We could improve that in future changes anyway

**Comment by arbulu89** [comment]
@diegoakechi I have done the changes. Anyway they are not visible here. As we changed the repo owner, and I did the changes after that in my fork, this PR is not gathering them. What we can do is merge this PR, and after that I could create the new PR from my fork with the changes. What do you think?

**Comment by diegoakechi** [comment]
> @diegoakechi I have done the changes. Anyway they are not visible here. As we changed the repo owner, and I did the changes after that in my fork, this PR is not gathering them. What we can do is merge this PR, and after that I could create the new PR from my fork with the changes. What do you think?

Works for me.

**Pull request merged**